docs/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.agent/index.md

```diff
@@ -0,0 +1,57 @@
+    n8n recommends using this agent for most use cases. It will handle most scenarios and provides the best experience when working with [tools](/advanced-ai/examples/understand-tools/).
+    /// 
+
+* [**Conversational Agent**](/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.agent/conversational-agent/): This agent has human-like conversations. It can maintain context, understand user intent, and provide relevant answers. This agent is typically used for building chatbots, virtual assistants and customer support systems. If your preferred AI model doesn't support tool calling or you're handling simpler interactions, this agent is a good option. 
+* [**OpenAI Functions Agent**](/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.agent/openai-functions-agent/): Use this agent with an [OpenAI functions model](https://platform.openai.com/docs/guides/gpt/function-calling){:target=_blank .external-link}. You must use the [OpenAI Chat Model](/integrations/builtin/cluster-nodes/sub-nodes/n8n-nodes-langchain.lmchatopenai/) with this agent.
+* [**Plan and Execute Agent**](/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.agent/plan-execute-agent/): This agent creates a high-level plan to solve the given task and then executes the plan step by step. It's most useful for tasks that require a structured approach and careful planning.
+* [**ReAct Agent**](/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.agent/react-agent/): This agent reasons about a given task, determines the necessary actions, and then executes them. It follows the cycle of reasoning and acting until it completes the task. The ReAct agent can break down complex tasks into smaller sub-tasks, prioritise them, and execute them one after the other.
+
+    /// note | No memory
+    The ReAct agent doesn't support memory sub-nodes. This means it can't recall previous prompts or simulate an ongoing conversation.
+    ///
+
+* [**SQL Agent**](/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.agent/sql-agent/): This agent uses a SQL database as a data source. It can understand natural language questions, convert them into SQL queries, execute the queries, and present the results in a user-friendly format. This agent is valuable for building natural language interfaces to databases.
+
+## Templates and examples
+<!-- see https://www.notion.so/n8n/Pull-in-templates-for-the-integrations-pages-37c716837b804d30a33b47475f6e3780 -->
+[[ templatesWidget(title, 'agent') ]]
+
+## Related resources
+
+Refer to [LangChain's documentation on agents](https://js.langchain.com/docs/modules/agents/agent_types/){:target=_blank .external-link} for more information about the service.
+
+New to AI Agents? Read the [n8n blog introduction to AI agents](https://blog.n8n.io/ai-agents/){:target=_blank .external-link}.
+
+--8<-- "_snippets/integrations/builtin/cluster-nodes/langchain-overview-link.md"
+--8<-- "_glossary/ai-glossary.md"
```

Comment:
With the way this ends up rendered, it might make sense to intro the glossary entries a bit so that it makes sense what the bullets here are.